versioned/persist/adapter/src/main/java/org/projectnessie/versioned/persist/adapter/spi/DbObjectsSerializers.java

```diff
@@ -0,0 +1,273 @@
+        new Deserializers.Base() {
+          @Override
+          public JsonDeserializer<?> findBeanDeserializer(
+              JavaType type, DeserializationConfig config, BeanDescription beanDesc) {
+            if (type.isTypeOrSubTypeOf(Hash.class)) {
+              return HASH_DESERIALIZER;
+            }
+            if (type.isTypeOrSubTypeOf(ContentsId.class)) {
+              return CONTENTS_ID_DESERIALIZER;
+            }
+            if (type.isTypeOrSubTypeOf(ByteString.class)) {
+              return BYTE_STRING_DESERIALIZER;
+            }
+            if (type.isTypeOrSubTypeOf(NamedRef.class)) {
+              return NAMED_REF_DESERIALIZER;
+            }
+            if (type.isTypeOrSubTypeOf(Key.class)) {
+              return KEY_DESERIALIZER;
+            }
+            return null;
+          }
+        };
+
+    Serializers serializers =
+        new Serializers.Base() {
+          @Override
+          public JsonSerializer<?> findSerializer(
+              SerializationConfig config, JavaType type, BeanDescription beanDesc) {
+            if (type.isTypeOrSubTypeOf(Hash.class)) {
+              return HASH_SERIALIZER;
+            }
+            if (type.isTypeOrSubTypeOf(ContentsId.class)) {
+              return CONTENTS_ID_SERIALIZER;
+            }
+            if (type.isTypeOrSubTypeOf(ByteString.class)) {
+              return BYTE_STRING_SERIALIZER;
+            }
+            if (type.isTypeOrSubTypeOf(NamedRef.class)) {
+              return NAMED_REF_SERIALIZER;
+            }
+            if (type.isTypeOrSubTypeOf(Key.class)) {
+              return KEY_SERIALIZER;
+            }
+            return null;
+          }
+        };
+
+    SimpleModule module =
+        new SimpleModule() {
+          @Override
+          public void setupModule(SetupContext context) {
+            super.setupModule(context);
+            context.addDeserializers(deserializers);
+            context.addSerializers(serializers);
+          }
+        };
+
+    return mapper.registerModule(module);
+  }
+
+  /**
+   * Convert a {@link #flattenKey(Key)} representation back into a {@link Key} object.
+   *
+   * @param flattened the {@link #flattenKey(Key)} representation
+   * @return parsed {@link Key}
+   */
+  public static Key fromFlattened(String flattened) {
+    ImmutableKey.Builder b = ImmutableKey.builder();
+    StringBuilder sb = new StringBuilder();
+    int l = flattened.length();
+    for (int p = 0; p < l; p++) {
+      char c = flattened.charAt(p);
+      if (c == '\\') {
+        c = flattened.charAt(++p);
+        sb.append(c);
+      } else if (c == '.') {
+        if (sb.length() > 0) {
+          b.addElements(sb.toString());
+          sb.setLength(0);
+        }
+      } else {
+        sb.append(c);
+      }
+    }
+    if (sb.length() > 0) {
+      b.addElements(sb.toString());
+      sb.setLength(0);
+    }
+    return b.build();
+  }
+
+  /**
+   * Flatten this {@link Key} into a single string.
```

Comment:
why deviate from the other parts of the code that serialize . as \00?

Reply:
The thought here was that `\000` is maybe not a "nice" character in a `VARCHAR` column